File: PrimesDeterminerWithList.py
print("Press Ctrl + C to end any long-running primes")
testNumber = int(input("Please give the desired number:"))
divisor = 2
factorable = False
factorsList = list()
n = 0
while(testNumber>divisor) : 
    if testNumber%divisor == 0 :
        factorsList.insert(n,divisor)
        n += 1
        print("Thinking....")
    divisor += 1
if len(factorsList) == 0 :
    print(testNumber, "is prime")
else :
    print(testNumber,"is not prime with factors of", factorsList)

File: PrimesDeterminerWithList.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class PrimesDeterminerWithList {
    private static double testNumber;
    private static double divisor;
    private static double result;
    private static boolean factorable;

    public static void main(String[] args){
        System.out.println("Please input your number:");
        Scanner sc = new Scanner(System.in);
        testNumber = sc.nextDouble();
        divisor = 2;
        factorable = false;
        List<Double> factorsList = new ArrayList();
        while(testNumber>divisor){
            result = testNumber/divisor;
            if (result == Math.round(result)){
                factorsList.add(divisor);
                System.out.println("Thinking...");
            }
            divisor = divisor+1;
        }
        if (factorsList.isEmpty() == true){
            System.out.println(testNumber + " is a prime number");
        }
        else{
            System.out.println(testNumber + " is not a prime number with factors of " + factorsList.toString() );
        }
    }
}
